exporter/collector/metrics.go

```diff
@@ -952,6 +964,31 @@ func numberDataPointToValue(
 		metricpb.MetricDescriptor_DOUBLE
 }
 
+// This function maps the values of the metric to certain types which are supported in Google Cloud Monitoring, but not in OTEL.
+// Such types includes STRING and BOOL. The conversion only happens for metric kind GAUGE and only under presence of certain attributes in the metric attributes.
+// The function returns the converted value and type if conditions are met, otherwise a nil value with value type MetricDescriptor_VALUE_TYPE_UNSPECIFIED is returned.
+func convertMetricKindToSupportedGCMTypes(metricKind metricpb.MetricDescriptor_MetricKind, pointLabels map[string]any) (*monitoringpb.TypedValue, metricpb.MetricDescriptor_ValueType) {
+	customType, customTypeOk := pointLabels[gcpCustomType]
+	customValue, customValueOk := pointLabels[gcpCustomValue]
+	if customTypeOk && customValueOk && metricKind == metricpb.MetricDescriptor_GAUGE {
```

Comment:
I'd recommend an early return here instead to avoid nesting. [see here](http://shortn/_didLEwCWAI)